pymatgen/analysis/Fstar/fstar.py

```diff
@@ -0,0 +1,246 @@
+            C : 0.0 0.0 14.275
+        PeriodicSite: Li:0.990, Ni:0.010 (0.0000, 0.0000, 0.0000) [0.0000, 0.0000, 0.0000]
+        PeriodicSite: Li:0.990, Ni:0.010 (1.4255, 0.8230, 4.7583) [0.6667, 0.3333, 0.3333]
+        PeriodicSite: Li:0.990, Ni:0.010 (-0.0000, 1.6460, 9.5167) [0.3333, 0.6667, 0.6667]
+        PeriodicSite: Li:0.010, Mn:0.333, Co:0.333, Ni:0.323 (0.0000, 0.0000, 7.1375) [0.0000, 0.0000, 0.5000]
+        PeriodicSite: Li:0.010, Mn:0.333, Co:0.333, Ni:0.323 (1.4255, 0.8230, 11.8958) [0.6667, 0.3333, 0.8333]
+        PeriodicSite: Li:0.010, Mn:0.333, Co:0.333, Ni:0.323 (-0.0000, 1.6460, 2.3792) [0.3333, 0.6667, 0.1667]
+        PeriodicSite: O (0.0000, 0.0000, 3.5688) [0.0000, 0.0000, 0.2500]
+        PeriodicSite: O (0.0000, 0.0000, 10.7063) [0.0000, 0.0000, 0.7500]
+        PeriodicSite: O (1.4255, 0.8230, 8.3270) [0.6667, 0.3333, 0.5833]
+        PeriodicSite: O (1.4255, 0.8230, 1.1895) [0.6667, 0.3333, 0.0833]
+        PeriodicSite: O (-0.0000, 1.6460, 13.0855) [0.3333, 0.6667, 0.9167]
+        PeriodicSite: O (-0.0000, 1.6460, 5.9480) [0.3333, 0.6667, 0.4167]
+
+        results in
+        labels - ['[0. 0. 0.]Li', '[0.  0.  0.5]Co', '[0.   0.   0.25]O']
+        '[0. 0. 0.]Li' - PeriodicSite: Li:0.990, Ni:0.010 (0.0000, 0.0000, 0.0000) [0.0000, 0.0000, 0.0000]
+        '[0.  0.  0.5]Co' - PeriodicSite: Li:0.010, Mn:0.333, Co:0.333, Ni:0.323
+            (0.0000, 0.0000, 7.1375) [0.0000, 0.0000, 0.5000]
+        '[0.   0.   0.25]O' - PeriodicSite: O (0.0000, 0.0000, 3.5688) [0.0000, 0.0000, 0.2500]
+        """
+
+        site_labels_fin = []
+        for ind1, struct in enumerate(self._equiv_inds):
+            site_labels = []
+            for ind2, site in enumerate(struct):
+                label = str(self._structures[ind1][site[0]].frac_coords) + \
+                        [str(sp) for sp, occ in self._structures[ind1][site[0]].species_and_occu.items()][0]
+                if label not in site_labels:
+                    site_labels.append(label)
+            if len(site_labels) > len(site_labels_fin):
+                site_labels_fin = site_labels
+        return site_labels_fin
+
+    def get_fstar_coords(self):
+
+        """
+        Calculate the f* coordinates for the list of structures.
+        """
+
+        fstar_df_full = pd.DataFrame(columns=self.site_labels)
+
+        for ind1, struct in enumerate(self._equiv_inds):
+            fstar_df = pd.DataFrame(columns=self.site_labels, data=[[0.0 for i in self.site_labels]])
+            for ind2, site in enumerate(struct):
+                occ_f_list = []
+                mult = len(site)
+                site_frac_coord = str(self._structures[ind1][site[0]].frac_coords)
+                column = [label for label in self.site_labels if site_frac_coord in label]
+                elements_and_occupancies = self._symstructs[ind1][site[0]].species_and_occu.items()
+                for sp, occ in elements_and_occupancies:
+                    if self.cif_list:

```

Comment:
Likewise, there's a lot of CIF-specific functionality here, which is probably simpler if working from the Structure.

It looks like the purpose of lines 174:196 is to create a list of fractional co-ordinates normalized to the [0, 1] range? This can be done on the Structure object something like:

```
structure = Structure.from_sites(input_structure.sites, to_unit_cell=True)
```